src/AssociationHydrator.php

```diff
@@ -12,22 +12,11 @@
 
 final class AssociationHydrator
 {
-    /** @var EntityManagerInterface */
-    private $entityManager;
-
-    /** @var ClassMetadata */
-    private $classMetadata;
-
-    /** @var PropertyAccessor */
-    private $propertyAccessor;
-
     public function __construct(
-        EntityManagerInterface $entityManager,
-        ClassMetadata $classMetadata,
-        ?PropertyAccessor $propertyAccessor = null
+        private readonly EntityManagerInterface $entityManager,
+        private readonly ClassMetadata $classMetadata,
+        private ?PropertyAccessor $propertyAccessor = null
```

Comment:
```suggestion
       ?PropertyAccessor $propertyAccessor = null
```
I'm not sure if we should label it as private since we modify it later in the constructor.

Reply:
??

Reply:
https://github.com/SyliusLabs/AssociationHydrator/pull/13/files#diff-d7407679d416107e0da59d1ae646307b7c4791f2bd5030c4918b64188660c1b6R20
Here's the second assignment

Reply:
The property promotion, in this case 'private', involves variable assignment under the hood, followed by the second assignment action inside constructor. It's a minor redundancy.

Reply:
I know, but if we leave it like it was, we have to declare property by hand anyway

Reply:
I've reverted this one 😃